core/base/model/AliceSkill.py

```diff
@@ -39,51 +40,87 @@ def __init__(self, supportedIntents: Iterable = None, authOnlyIntents: dict = No
 		self._required = False
 		self._databaseSchema = databaseSchema
 		self._widgets = dict()
+		self._intentsDefinitions = dict()
 
-		if not supportedIntents:
-			supportedIntents = list()
+		self._supportedIntents: Dict[str, Intent] = self.buildIntentList(supportedIntents)
+		self.loadIntentsDefinition()
 
-		self._myIntents: List[Intent] = list()
-		self._supportedIntents: Dict[str, Tuple[(str, Intent), Callable]] = dict()
-		for item in (*supportedIntents, *self.intentMethods()):
-			if isinstance(item, tuple):
-				self._supportedIntents[str(item[0])] = item
-				self._myIntents.append(item[0])
-			elif isinstance(item, Intent):
-				self._supportedIntents[str(item)] = (item, self.onMessage)
-				self._myIntents.append(item)
-			elif isinstance(item, str):
-				self._supportedIntents[item] = (item, self.onMessage)
-
-		self._authOnlyIntents: Dict[str, AccessLevel] = {str(intent): level.value for intent, level in authOnlyIntents.items()} if authOnlyIntents else dict()
 		self._utteranceSlotCleaner = re.compile('{(.+?):=>.+?}')
 		self.loadWidgets()
 
 
-	@classmethod
-	def decoratedIntentMethods(cls):
-		from core.util.Decorators import IntentHandler
-		for name in dir(cls):
-			method = getattr(cls, name)
-			while isinstance(method, IntentHandler.Wrapper):
-				yield method
-				method = method.decoratedMethod
+	def loadIntentsDefinition(self):
+		dialogTemplate = Path(self.getCurrentDir(), 'dialogTemplate')
+
+		for lang in self.LanguageManager.supportedLanguages:
+			try:
+				path = dialogTemplate / f'{lang}.json'
+				if not path.exists():
+					continue
+
+				with path.open('r') as fp:
+					data = json.load(fp)
+
+					if not 'intents' in data:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [test for membership should be 'not in'](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4699682)